### PR TITLE
readme: Add basic coding-guidelines and commit template

### PR DESCRIPTION
**Pull request by EdDev**
For consistency I'm adding some basic guidelines.
These are taken from oVirt VDSM project.

### CODE REVIEW

File: commit-template.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+topic: Short summary under 50 chars (if possible)
+
```
Comment (tyll) [inline commit-template.txt]: How about quoting the seven rules for a good commit message from https://chris.beams.io/posts/git-commit/

IMHO at least using the imperative mood is missing here.

File: README.developer.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# NMState Coding Guidelines
+
+- NMState is written primarily in Python, and its coding style should follow the best practices of Python coding unless otherwise declared.
+- PEP8 is holy.
+- Tests are holy. Production code must be covered by unit tests and basic integration tests.
```
Comment (tyll) [inline README.developer.md]: We do not follow this completely, therefore "must" is a little bit too strong.

File: README.developer.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# NMState Coding Guidelines
+
+- NMState is written primarily in Python, and its coding style should follow the best practices of Python coding unless otherwise declared.
+- PEP8 is holy.
```
Comment (tyll) [inline README.developer.md]: How about mentioning that commits should be checked with tox here, it includes a pep8 check afaik.

File: README.developer.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  You can set the template for nmstate by configuring git: `git config commit.template commit-template.txt` 
+- Packages, modules, functions, methods and variables should use underscore_separated_names.
+- Class names are in CamelCase.
+- Imports should be grouped in the following order:
+-- Standard library imports
+-- Related third party imports
+-- Local application-specific or library-specific imports.
+- All indentation is made of the space characters. Tabs are evil. In makefiles, however, tabs are obligatory.
+  White space between code stanzas are welcome. They help to create breathing while reading long code. However, splitting stanzas into helper functions could be even better.
+
```
Comment (tyll) [inline README.developer.md]: The preference of single ticks should be mentioned here and ideally also some tooling to help to conform to this (or we use double quotes for which tooling exists ;-)).

File: README.developer.md (hunk 2 above)
Comment (cathay4t) [reply README.developer.md]: I am OK with `must`. But it could be better we could say more about how to create unit test and integration test like:
```
Please refer to commit <commit_id_with_unit_test_integration_test> as an example.
```

Comment (EdDev) [reply README.developer.md]: 'must': We should follow it, and I hope you will nack my patches if it does not. But I should rephrase it to: "unit tests and/or basic integration tests". There are tests which are better covered using the integration tests and not unit tests (the ones who need too much mocking).

About the 'how': Well, I do hope the one who attempts to post a change knows where too look. I usually go to tests to learn how to change something or how it currently works.

File: README.developer.md (hunk 4 above)
Comment (EdDev) [reply README.developer.md]: I got used to single quotes and it feels more minimalistic.. the python way. As we are already there, and I like it so much, I will indeed add it :)
I do not have tooling, but from experience after you get used to it, it sticks.

File: commit-template.txt (hunk 1 above)
Comment (EdDev) [reply commit-template.txt]: Ok, will try to embed it here.

**Comment by tyll** [comment]
@EdDev your caller to callee preference is also missing but this should also have tooling support.

**Comment by EdDev** [comment]
@tyll I have a bunch of other preferences in the queue, lets push this basic version and then continue with the others in a following patch.

I also like tooling, but I think it cannot be a condition for guidance and preferences. Some even cannot have tooling as exceptions are acceptable. With the caller-callee preference, it sometimes does not work with python due to the interpreter processing or because of an contradicting preference (one I can think of is: pytest prefixes).

**Comment by EdDev** [comment]
I opened https://nmstate.atlassian.net/browse/NMSTATE-70 to commit for doing this in this sprint.
I hope this is enough.
